docs/how-to-guides/develop-a-tool/add-a-tool-icon.md

```diff
@@ -14,40 +14,76 @@ Adding a custom tool icon is optional. If you do not provide one, the system use
-Here we use [an existing tool project](https://github.com/microsoft/promptflow/tree/main/examples/tools/tool-package-quickstart) as an example.
-```
-cd D:\proj\github\promptflow\examples\tools\tool-package-quickstart
 
-python D:\proj\github\promptflow\scripts\tool\generate_package_tool_meta.py -m my_tool_package.tools.my_tool_1 -o my_tool_package\yamls\my_tool_1.yaml -i my_tool_package\icons\custom-tool-icon.png
+PF CLI will copy the icon file to the folder `<package-name>/icon/<icon-file-name>` and generate a tool script in the package. The tool icon will be configured in the tool script. Here we use [an existing tool project](https://github.com/microsoft/promptflow/tree/main/examples/tools/tool-package-quickstart) as an example, the code is as follows:
+```python
+from pathlib import Path
+
+from promptflow import tool
+from promptflow.connections import CustomConnection
+
+
+@tool(
+    name="tool_name",
+    description="This is tool_name tool",
+    icon=Path(__file__).parent.parent / "icon" / <your-icon-file-name>,
+)
+def tool_name(connection: CustomConnection, input_text: str) -> str:
+    # Replace with your tool code.
+    # Usually connection contains configs to connect to an API.
+    # Use CustomConnection is a dict. You can use it like: connection.api_key, connection.api_base
+    # Not all tools need a connection. You can remove it if you don't need it.
+    return "Hello " + input_text
 ```
 
-In the auto-generated tool YAML file, the custom tool icon data URI is added in the `icon` field:
-```yaml
-my_tool_package.tools.my_tool_1.my_tool:
-  function: my_tool
-  icon: data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAACR0lEQVR4nKWS3UuTcRTHP79nm9ujM+fccqFGI5viRRpjJgkJ3hiCENVN/QMWdBHUVRdBNwX9ARHd2FVEWFLRjaS9XPmSC/EFTNOWc3Pi48y9PHNzz68L7UXTCvreHM65+PA953uElFLyHzLvHMwsJrnzfJqFeAan3cKV9mr8XseeAOXX5vqjSS53jdF+tIz1nIFAMDCzwpvJ5b87+LSYYHw+gcWkEAwluXnOR2Q1R+9YjJ7BKJG4zoXmqr0ddL3+QnV5EeUOK821LsJammcjEeZiafJScrd3bm8H6zkDd4mVztZKAK49/Mj8is4Z/35GPq9R5VJ5GYztDtB1HT1vovGQSiqVAqDugI3I6jpP3i9x9VQVfu8+1N/OvbWCqqqoBSa6h1fQNA1N0xiYTWJSBCZF8HgwSjQapbRQ2RUg5NYj3O6ZochmYkFL03S4mImIzjFvCf2xS5gtCRYXWvBUvKXjyEVeTN/DXuDgxsnuzSMK4HTAw1Q0hZba4NXEKp0tbpq9VkxCwTAETrsVwxBIBIYhMPI7YqyrtONQzSznJXrO4H5/GJ9LUGg0YFYydJxoYnwpj1s9SEN5KzZz4fYYAW6dr+VsowdFgamlPE/Hs8SzQZYzg0S+zjIc6iOWDDEc6uND+N12B9/VVu+mrd79o38wFCCdTeBSK6hxBii1eahxBlAtRbsDdmoiHGRNj1NZ7GM0NISvzM9oaIhiqwOO/wMgl4FsRpLf2KxGXpLNSLLInzH+CWBIA6RECIGUEiEUpDRACBSh8A3pXfGWdXfMgAAAAABJRU5ErkJggg==
-  inputs:
-    connection:
-      type:
-      - CustomConnection
-    input_text:
-      type:
-      - string
-  module: my_tool_package.tools.my_tool_1
-  name: my_tool
-  type: python
+The folder structure of the generated tool package is as follows:
 ```
+<package-name>
+│   MANIFEST.in
+│   README.md
+│   setup.py
+│
+├───icon
+│       <icon-file-name>
+│
+└───<package-name>
+        <tool-name>.py
+        utils.py
+        __init__.py
+```
+
+
+
+### Add an icon to an existing package tool
```

Comment:
Remove this.

Reply:
Move this part to the QA section: ### Can I add a tool icon to an existing tool package?